platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -649,7 +656,7 @@ def __init__(self):
         self.image_ext_name = self.COMPONENT_FIRMWARE_EXTENSION
         self.onie_updater = ONIEUpdater()
 
-    def __install_firmware(self, image_path):
+    def __install_firmware(self, image_path, reboot=False):
```

Comment:
@alexrallen this will break the original `non-autoupdate` behaviour

Reply:
Lets merge all your concerns about the original non-auto update behavior into this thread since essentially all other firmwares other than SSD have similar requirements. 

How exactly do you think this will break the behavior? I recognize that the "install" command will no longer automatically reboot the switch but I believe this is correct with regard to the HLD.

From discussions with MSFT they claim that "install" is only performing the actions up to any reboot behavior while "update" is performing all actions necessary to fully update the firmware (including a reboot if necessary). I have updated the flows to be compatible with this definition which is also a requirement for having the auto update flow work as it assumes our API implementation acts in this way.